src/context/ThemeContext.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { createContext, useContext, useEffect, useState } from 'react';
+
+import { ApiService } from '@/api';
+import type { ThemeData } from '@/api/types';
+
+interface ThemeContextType {
+  themes: ThemeData[];
+}
+
+const ThemeContext = createContext<ThemeContextType | undefined>(undefined);
+
+export const ThemeProvider = ({ children }: { children: React.ReactNode }) => {
+  const [themes, setThemes] = useState<ThemeData[]>([]);
+
+  useEffect(() => {
+    const fetchThemes = async () => {
+      try {
+        const response = await ApiService.fetchThemes();
+        setThemes(response.themes);
+      } catch (error) {
+        console.error('에러가 발생했어요.', error);
+      }
+    };
+
+    fetchThemes();
+  }, []);
+
+  return <ThemeContext.Provider value={{ themes }}>{children}</ThemeContext.Provider>;
+};
```

Comment:
```suggestion
export const ThemeProvider = ({ children, themes }: PropsWithChildren<{ themes: ThemeData[] }>) => {
  return <ThemeContext.Provider value={{ themes }}>{children}</ThemeContext.Provider>;
};
```
이렇게 표현해주고,

사용할 때는 이런 모습인거죠.
```tsx
const themes = useThemes();

<ThemeProvider themes={themes}></ThemeProvider>
```

만약 useThemes가 loading이나 error도 표현한다고 가정하면

```tsx
const { themes, error, loading } = useThemes();
```
이런식으로 표현할 수 있도록 만들어주고,

```tsx
<ThemeProvider {...useThemes()}></ThemeProvider>
```
이렇게 바로 props에다가 hooks을 집어넣을 수 있답니다! 

--

이렇게 하는 이유는 ThemeProvider가 한 가지 역할만 하도록 만들기 위함입니다!
theme 데이터를 제공만 하면 되지 그걸 꼭 ThemeProvider가 만들 필요는 없으니까요 ㅎㅎ